python/calico/felix/endpoint.py

```diff
@@ -887,16 +893,26 @@ def _apply_endpoint_update(self):
 
             tiers = pending_endpoint.get("tiers", [])
             tier_dict = OrderedDict()
+            raw_tier_dict = OrderedDict()
             for tier in tiers or []:
                 pols = []
+                raw_pols = []
                 for pol in tier["policies"] or []:
                     pol_id = TieredPolicyId(tier["name"], pol)
-                    pols.append(pol_id)
+                    if is_untracked(pol_id):
```

Comment:
I don't think using a global here is going to fly; the global might not have been updated when this code runs.  I think you'll have to plumb through the untracked-ness of the policies in the protobuf message.  I'd do that by adding an "untracked_tiers" field to the HostEndpoint object, then plumbing it through.